contracts/chain-adapters/Arbitrum_L3_Adapter.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { AdapterInterface } from "./interfaces/AdapterInterface.sol";
+
+/**
+ * @notice Contract containing logic to send messages from L1 to Arbitrum-like L3s using an intermediate L2 message forwarder.
+ * @notice This contract requires an L2 forwarder contract to be deployed, since we overwrite the target field to this new target.
+ * @dev Public functions calling external contracts do not guard against reentrancy because they are expected to be
+ * called via delegatecall, which will execute this contract's logic within the context of the originating contract.
+ * For example, the HubPool will delegatecall these functions, therefore its only necessary that the HubPool's methods
+ * that call this contract's logic guard against reentrancy.
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+contract Arbitrum_L3_Adapter is AdapterInterface {
+    address public immutable l2Adapter;
+    address public immutable l2Forwarder;
+
+    error RelayMessageFailed();
+    error RelayTokensFailed(address l1Token);
+
+    /**
+     * @notice Constructs new Adapter for sending tokens/messages to Arbitrum-like L3s.
+     * @param _l2Adapter Address of the adapter contract on mainnet which implements message transfers
+     * and token relays.
+     */
+    constructor(address _l2Adapter, address _l2Forwarder) {
+        l2Adapter = _l2Adapter;
+        l2Forwarder = _l2Forwarder;
```

Comment:
The naming of `l2Adapter` and `l2Forwarder` is a bit counter-intuitive because one is on the L1 and the other is on the L2. In the context of the chain adapters themselves, they're otherwise just called `adapter`; should we maybe follow that?

Reply:
I think that's reasonable. https://github.com/across-protocol/contracts/pull/607/commits/2daf57b25a2cff090b1feecbb4eadab1a707a81f